dns-controller/pkg/watchers/servicecontroller.go

```diff
@@ -141,21 +141,34 @@ func (c *ServiceController) updateServiceRecords(service *v1.Service) {
 	specExternal := service.Annotations[AnnotationNameDnsExternal]
 	if specExternal != "" {
 		var ingresses []dns.Record
-		for i := range service.Status.LoadBalancer.Ingress {
-			ingress := &service.Status.LoadBalancer.Ingress[i]
-			if ingress.Hostname != "" {
-				// TODO: Support ELB aliases
-				ingresses = append(ingresses, dns.Record{
-					RecordType: dns.RecordTypeCNAME,
-					Value:      ingress.Hostname,
-				})
-			}
-			if ingress.IP != "" {
-				ingresses = append(ingresses, dns.Record{
-					RecordType: dns.RecordTypeA,
-					Value:      ingress.IP,
-				})
+
+		if service.Spec.Type == v1.ServiceTypeLoadBalancer {
+			for i := range service.Status.LoadBalancer.Ingress {
+				ingress := &service.Status.LoadBalancer.Ingress[i]
+				if ingress.Hostname != "" {
+					// TODO: Support ELB aliases
+					ingresses = append(ingresses, dns.Record{
+						RecordType: dns.RecordTypeCNAME,
+						Value:      ingress.Hostname,
+					})
+					glog.V(4).Infof("Found CNAME for service %s/%s: %q", service.Namespace, service.Name, ingress.Hostname)
+				}
+				if ingress.IP != "" {
+					ingresses = append(ingresses, dns.Record{
+						RecordType: dns.RecordTypeA,
+						Value:      ingress.IP,
+					})
+					glog.V(4).Infof("Found A for service %s/%s: %q", service.Namespace, service.Name, ingress.IP)
+				}
 			}
+		} else if service.Spec.Type == v1.ServiceTypeNodePort {
+			ingresses = append(ingresses, dns.Record{
+				RecordType: dns.RecordTypeAlias,
+				Value:      dns.AliasForNodesInRole("node"),
+			})
+			glog.V(4).Infof("Setting internal alias for NodePort service %s/%s", service.Namespace, service.Name)
+		} else {
+			glog.V(2).Infof("Cannot expose service %s/%s of type %q", service.Namespace, service.Name, service.Spec.Type)
```

Comment:
Looks like this function doesn't support returning `errors`.. Do we need that here? Or is a log message sufficient when we can't expose a service?

If this is fine - Should we bump the message higher than `V(2)`?